hmftools-build.py

```diff
@@ -39,9 +39,22 @@ def set_version(self, version):
     @staticmethod
     def deploy_all(*modules):
         module_str = ','.join([m.name for m in modules])
-        subprocess.run(['mvn', 'deploy', '-B', '-pl', module_str, '-am', '-DdeployAtEnd=true'], check=True)
+        #subprocess.run(['mvn', 'deploy', '-B', '-pl', module_str, '-am', '-DdeployAtEnd=true'], check=True)
+        subprocess.run(['mvn', 'install', '-B', '-pl', module_str, '-am'], check=True)
 
 
+class Docker:
+    def __init__(self, module, version):
+        self.module = module
+        self.version = version
+        self.internal_image = f'europe-west4-docker.pkg.dev/hmf-build/hmf-docker/{self.module}:{self.version}'
+        self.external_image = f'europe-west4-docker.pkg.dev/hmf-build/build-registry-docker/{self.module}:{self.version}'
```

Comment:
So if I understand correctly, in your approach you're choosing to release to an open-to-the-world artifact registry? Did you also consider releasing to docker central? (I think this approach is fine, but it could be good to also consider releasing to docker central for visibility.)

Reply:
Haven't quite pinned down the external repo, I'd be just fine with Docker central as well, but the principal requirement here was that Stephen (Watts, in Melbourne, who has built the Nextflow implementation) would be able to access them. I kicked the can down the road a bit thinking I'd just put this in and deal with the details of exactly where it is going to go once we've got everybody on board, but maybe I'll give it a bit more thought up-front to bolster the case for this approach.

Reply:
I think we'll start with pushing internally and to DockerHub. DockerHub will be accessible to NextFlow. I guess the "external" image above is just a placeholder?

Bioconda, which is currently used by Stephen requires a separate "recipe" and approval process for each change, so let's consider it a non-starter for now.